department-of-veterans-affairs/caseflow
# Add "cancel convert hearing type to virtual" action

Resolves #15507 

### Description

- Add new cancel action for `ConvertHearingRequestTypeTask`
- Refactor `CancelTaskModal` to make it functional
- Add option to hide the instructions text field on `CancelTaskModal`
- Add jest test and feature test
- Add stories for `CancelTaskModal`

### Acceptance Criteria
- [x] Users with the "Admin Intake", "Build HearSched", or "Edit HearSched" tasks are given the to option to cancel a ChangeHearingRequestTypeTask
- [x] When a Case Details page is loaded for an appeal that has a cancelled ChangeHearingRequestTypeTask in its task tree, a new ChangeHearingRequestTypeTask is not created

### Testing Plan
1. Login as BVASYELLOW
2. Navigate to http://localhost:3000/queue/appeals/1986897
3. Ensure there's a cancel action
4. Complete the cancel action
5. Try reloading the page

### User Facing Changes

---

<img width="1436" alt="Screen Shot 2020-10-30 at 4 04 30 PM" src="https://user-images.githubusercontent.com/1298274/97752026-aaba0b00-1ac9-11eb-8e70-7c86d8931e56.png">

---

<img width="1436" alt="Screen Shot 2020-10-30 at 4 04 36 PM" src="https://user-images.githubusercontent.com/1298274/97752030-abeb3800-1ac9-11eb-8634-32a8a508abf2.png">

---

<img width="1436" alt="Screen Shot 2020-10-30 at 4 04 44 PM" src="https://user-images.githubusercontent.com/1298274/97752032-abeb3800-1ac9-11eb-9e60-6114536b5480.png">


### Storybook Story
*For Frontend (Presentationa) Components*
